### PR TITLE
Adapts checksum functionality to RSKIP60 and EIP55

### PR DESCRIPTION
**Pull request by alepc253**
## Description

This PR adapts checksum functionality to [RSKIP-60](https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP60.md) compatible with  [EIP-55](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md).
The implementation considers sending optional `chainId` parameter to checksum methods (with `chainId = null` as default, Ethereum checksum is applied).

Benefit

- Allows checksum implementation in any network.
- Can distinguish between testnets and mainnets.

## Type of change

- [ ] Bug fix 
- [X] New feature 
- [ ] Breaking change
- [X] Enhancement

## Checklist:

- [X] I have selected the correct base branch.
- [X] I have performed a self-review of my own code.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have made corresponding changes to the documentation.
- [X] My changes generate no warnings.
- [X] I have updated or added types for all modules I've changed
- [X] Any dependent changes have been merged and published in downstream modules.
- [X] I ran ```npm run test``` in the root folder with success and extended the tests if necessary.
- [X] I ran ```npm run build``` in the root folder and tested it in the browser and with node.
- [X] I ran ```npm run dtslint``` in the root folder and tested that all my types are correct
- [ ] I have tested my code on an ethereum test network.

### CODE REVIEW

**Review by nivida — CHANGES_REQUESTED** [review]
Thanks for submitting this PR! 

Could you update the related documentation and the types for both methods?

File: docs/_build/html/_sources/web3-utils.rst.txt [hunk 1]
```diff
@@ -418,6 +418,7 @@ Parameters
 ----------
 
 1. ``address`` - ``String``: An address string.
+2. ``chainId`` - ``number`` (optional): Chain id where checksummed address should be valid, defaults to ``null``.
```
Comment (nivida) [inline docs/_build/html/_sources/web3-utils.rst.txt]: Thanks for extending the documentation! Could you add here the URL to the RSKIP-60 spec?

File: docs/_build/html/_sources/web3-utils.rst.txt [hunk 2]
```diff
@@ -482,6 +487,42 @@ Example
     web3.utils.toChecksumAddress('0XC1912FEE45D61C87CC5EA59DAE31190FFFFF232D');
     > "0xc1912fEE45d61C87Cc5EA59DaE31190FFFFf232d" // same as above
 
+    web3.utils.toChecksumAddress('0x5aaeb6053f3e94c9b9a09f33669435e7ef1beaed', 30);
+    > "0x5aaEB6053f3e94c9b9a09f33669435E7ef1bEAeD"
+
+------------------------------------------------------------------------------
+
+
+stripHexPrefix
```
Comment (nivida) [inline docs/_build/html/_sources/web3-utils.rst.txt:496]: Could you add the required test cases in the web3-utils module for this method?

File: docs/_build/html/_sources/web3-utils.rst.txt [hunk 3]
```diff
@@ -500,6 +541,7 @@ Parameters
 ----------
 
 1. ``address`` - ``String``: An address string.
+2. ``chainId`` - ``number`` (optional): Chain id where checksummed address should be valid, defaults to ``null``.
```
Comment (nivida) [inline docs/_build/html/_sources/web3-utils.rst.txt]: Could you add here the URL to the RSKIP-60 spec?

File: packages/web3-utils/src/Utils.js [hunk 4]
```diff
@@ -107,30 +109,44 @@ export const isAddress = (address) => {
...
-            (parseInt(addressHash[i], 16) > 7 && address[i].toUpperCase() !== address[i]) ||
-            (parseInt(addressHash[i], 16) <= 7 && address[i].toLowerCase() !== address[i])
-        ) {
+export const checkAddressChecksum = (address, chainId = null) => {
+    const stripAddress = stripHexPrefix(address).toLowerCase()
+    const prefix = chainId != null ? (chainId.toString() + '0x') : ''
+    const keccakHash = Hash.keccak256(prefix + stripAddress).toString('hex').replace(/^0x/i, '');
+
+    for (let i = 0; i < stripAddress.length; i++) {
+        let output = parseInt(keccakHash[i], 16) >= 8 ?
```
Comment (nivida) [inline packages/web3-utils/src/Utils.js]: Could you change this line to: 
`` let output = parseInt(keccakHash[i], 16) >= 8 ?
            stripAddress[i].toUpperCase() :
            stripAddress[i].toLowerCase();
``

File: packages/web3-utils/src/index.js [hunk 5]
```diff
@@ -240,31 +241,32 @@ export const toWei = (number, unit) => {
...
-    for (let i = 0; i < address.length; i++) {
-        // If ith character is 9 to f then make it uppercase
-        if (parseInt(addressHash[i], 16) > 7) {
-            checksumAddress += address[i].toUpperCase();
-        } else {
-            checksumAddress += address[i];
-        }
-    }
+    for (let i = 0; i < stripAddress.length; i++)
+        checksumAddress += parseInt(keccakHash[i], 16) >= 8 ?
```
Comment (nivida) [inline packages/web3-utils/src/index.js]: Could you change this line to: 
`` let output = parseInt(keccakHash[i], 16) >= 8 ?
            stripAddress[i].toUpperCase() :
            stripAddress[i].toLowerCase();
``

**Review by nivida — CHANGES_REQUESTED** [review]

File: packages/web3-utils/types/index.d.ts [hunk 6]
```diff
@@ -149,7 +150,7 @@ export interface Utils {
     randomHex(bytesSize: number): string;
     utf8ToHex(string: string): string;
     stringToHex(string: string): string;
-    toChecksumAddress(address: string): string;
+    toChecksumAddress(address: string, chainId?: number): string;
```
Comment (nivida) [inline packages/web3-utils/types/index.d.ts:153]: Could you also extend the type tests for all these changes?

**Review by nivida — CHANGES_REQUESTED** [review]

**Comment by alepc253** [comment]
@nivida thanks! I'm working on your reviews.
One important aspect to consider (and still didn't boarded) is the method `inputAddressFormatter` from `web3-core-helpers/src/Formatters.js`. `chainId` should be added to that formatter to be consistent to this change. I guess it would be very helpful to have `chainId` as a global parameter but I don't know if that is possible at this level... what do you think?

**Comment by nivida** [comment]
@alepc253 I would like to write that down in a new issue and to implement it later as an enhancement. The question will be if Web3 should automatically load the ``chainId`` from the connected node if no ``chainId`` got defined from the developer.

**Review by nivida — APPROVED** [review]

**Pull request merged**